### PR TITLE
Improves functions data & signature encode methods

### LINKED ISSUE #184: Replace encodeFunctionCall usages with methods.myMethod.encodeABI
Please, replace `eth.abi.encodeFunctionCall` usages in openst-contracts (`grep` the repo to find occurrences) with `methods.myMethod.encodeABI` calls (https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#methods-mymethod-encodeabi).

### PR DESCRIPTION
**Pull request by pgev**
- Replaces web3.eth.abi.encodeFunctionCall -s by
    MyContract.methods.MyMethod(...).encodeABI() -s.
  - Replaces web3.eth.abi.encodeFunctionSignature() -s by
    MyContract.jsonInterface.abi.methods.MyMethod.signature -s.
  - generateSwapOwnerData and generateCoGatewayRedeemFunctionData
    functions explicitly use corresponding functions' abis as contracts
    of those functions are out of the current project.
  - Updates all (almost) modules within package.json, adjusts
    package-lock.json, fixes versions (no ^) of modules within
    package.json.
  - Increases Truffle's solc version from '0.5.0' to '0.5.8'.
  - Fixes typo: DelayedRedcoveryModuleCreated: removes `d`.
  - Fixes typo: RECOERY_MODULE_DOMAIN_SEPARATOR_TYPEHASH: add `V`.

Resolves: #184

### CODE REVIEW

**Comment by abhayks1** [comment]
Reviewing 🚀

File: package.json [hunk 1]
```diff
@@ -41,28 +41,24 @@
...
+        "bn.js": "4.11.8",
+        "eslint": "5.16.0",
+        "eslint-config-airbnb-base": "13.1.0",
+        "eslint-plugin-import": "2.17.2",
+        "ethereumjs-util": "6.1.0",
+        "ethlint": "1.2.4",
+        "ganache-cli": "6.4.3",
+        "npm-run-all": "4.1.5",
+        "solium": "1.2.4",
+        "web3": "1.0.0-beta.51",
```
Comment (abhayks1) [inline package.json:59]: We can update it to 1.0.0-beta.55 if no major test issue.
https://www.npmjs.com/package/web3

**Review by abhayks1 — DISMISSED** [review]
Nice work 👍 💯 
Added a small comment. It will be nice if that's handled as we are updating package.json npm versions anyways. Let me know if there are test failures 🔴

File: package.json (hunk 1 above)
Comment (pgev) [reply package.json:59]: Sure. Could you please create a ticket for that. Thx.

Comment (pgev) [reply package.json:59]: https://github.com/OpenST/openst-contracts/issues/206

**Comment by schemar** [comment]
Checking 🦑

**Review by schemar — APPROVED** [review]
Nice clean-up 🐴

**Pull request merged**